addons/default/visiosoft/advs-module/src/Adv/AdvRepository.php

```diff
@@ -91,15 +91,25 @@ public function searchAdvs($type, $param = null, $customParameters = null, $limi
                 });
             }
         }
-        $country = isset($param['country']) ?
-            $param['country'] : setting_value('visiosoft.module.location::default_country');
-        if ($country) {
-            $query = $query->where('country_id', $country);
-        }
-        if ($city) {
-            $query = $query->where('city', $city->id);
-        } elseif (isset($param['city']) and !empty($param['city']) and !empty(array_filter($param['city']))) {
-            $query = $query->whereIn('city', $param['city']);
+        if (!setting_value('visiosoft.module.location::hide_location_filter')) {
+            $country = isset($param['country']) ? $param['country'] : setting_value('visiosoft.module.location::default_country');
+            if ($country) {
+                $query = $query->where('country_id', $country);
+            }
+            if ($city) {
+                $query = $query->where('city', $city->id);
+            } elseif (isset($param['city']) and !empty($param['city']) and !empty(array_filter($param['city']))) {
+                $query = $query->whereIn('city', $param['city']);
+            }
+            if (isset($param['district']) and !empty(array_filter($param['district']))) {
```

Comment:
if we need this part `!empty($param['city'])` why are we not adding it here as well